api/valor_api/backend/metrics/text_generation.py

```diff
@@ -0,0 +1,529 @@
+)
+from valor_api.backend.metrics.metric_utils import (
+    create_metric_mappings,
+    get_or_create_row,
+    log_evaluation_duration,
+    log_evaluation_item_counts,
+    validate_computation,
+)
+from valor_api.backend.query import generate_query
+from valor_api.enums import MetricType
+
+LabelMapType = list[list[list[str]]]
+
+LLM_GUIDED_METRICS = {
+    "AnswerRelevance",
+    "Coherence",
+}
+
+TEXT_COMPARISON_METRICS = {"BLEU", "ROUGE"}
+
+
+def _calculate_rouge_scores(
+    predictions: str | list[str],
+    references: list[str],
+    rouge_types: list[str] = ["rouge1", "rouge2", "rougeL", "rougeLsum"],
+    use_stemmer: bool = False,
+) -> list[dict]:
+    """
+    Calculate ROUGE scores for a prediction (or list of predictions) given some set of references.
+
+    Parameters
+    ----------
+    prediction: str | list[str]
+        The prediction (or list of predictions) to score. Each prediction should be a string with tokens separated by spaces.
+    references: list[str] | list[list[str]]
+        A list of reference for a given prediction. Each reference should be a string with tokens separated by spaces.
+    rouge_types: list[str]
+        A list of rouge types to calculate. Defaults to ['rouge1', 'rouge2', 'rougeL', 'rougeLsum'], where `rouge1` is unigram-based scoring, `rouge2` is bigram-based scoring, `rougeL` is scoring based on sentences (i.e., splitting on "." and ignoring "\n"), and `rougeLsum` is scoring based on splitting the text using "\n".
+    use_stemmer: bool
+        If True, uses Porter stemmer to strip word suffixes. Defaults to False.
+
+    Raises
+    ----------
+    ValueError
+        If prediction is neither a string nor a list.
+    """
+    if not predictions or not references or isinstance(references, str):
+        raise ValueError(
+            "Received incorrect inputs. predictions should be a string, references a list of strings, and weights a list/tuple of floats"
+        )
+
+    rouge = evaluate.load("rouge")
+
+    # handle case where user passes in a single prediction
+    if isinstance(predictions, str):
+        processed_prediction = [predictions]
+        processed_references = [references]
+        # handle case where user passes multiple predictions
+    elif isinstance(predictions, list) and all(
+        [isinstance(lst, list) for lst in references]
+    ):
+        processed_prediction = predictions
+        processed_references = references
+    else:
+        raise ValueError(
+            "prediction should be a str or list[str]. If prediction is a list[str], then references must be a list of lists."
+        )
+
+    metrics = rouge.compute(
+        predictions=processed_prediction,
+        references=processed_references,
+        rouge_types=rouge_types,
+        use_stemmer=use_stemmer,
+        use_aggregator=False,  # aggregation gives us an average across all predicitons, which isn't what we want
+    )
+
+    assert metrics is not None  # handle type error
```

Comment:
`assert` should not be used in production code as it can be ignored depending on python interpreter flags. See [link](https://stackoverflow.com/questions/944592/best-practice-for-using-assert).

I would suggest changing this to `if metrics is None: raise TypeError(message)`